OBSDOCS-1546: Logging Z-Stream Release Notes - 6.1.1 (#86343)
openshift/openshift-docs: modules/log6x-6-1-1-rn.adoc

// Module included in the following assemblies:
//
// * observability/logging/logging-6.1/log6x-release-notes-6.1.adoc

:_mod-docs-content-type: REFERENCE
[id="logging-release-notes-6-1-1_{context}"]
= Logging 6.1.1 Release Notes

This release includes link:https://access.redhat.com/errata/RHBA-2024:10992[{logging-uc} {for} Bug Fix Release 6.1.1].

[id="logging-release-notes-6-1-1-enhancements_{context}"]
== New Features and Enhancements

* With this update, the {loki-op} supports configuring the workload identity federation on the {gcp-first} by using the Cluster Credential Operator (CCO) in {product-title} 4.17 or later. (link:https://issues.redhat.com/browse/LOG-6420[LOG-6420])

[id="logging-release-notes-6-1-1-bug-fixes_{context}"]
== Bug Fixes

* Before this update, when any infrastructure namespaces were specified in the application inputs in the `ClusterLogForwarder` custom resource, logs were generated with the incorrect `log_type: application` tags. With this update, when any infrastructure namespaces are specified in the application inputs, logs are generated with the correct `log_type: infrastructure` tags. (link:https://issues.redhat.com/browse/LOG-6373[LOG-6373])

> * Before this update, the collector was discarding longer audit log messages with the following error message: *Internal log [Found line that exceeds max_line_bytes; discarding.]*. With this update, the discarding of longer audit messages is avoided by increasing the audit configuration thresholds: The maximum line size, `max_line_bytes`, is `3145728` bytes. The maximum number of bytes read during a read cycle, `max_read_bytes`, is `262144` bytes. (link:https://issues.redhat.com/browse/LOG-6379[LOG-6379])

Reviewer: It's probably correct according to some style guides but I feel the capitalization after the colon is a little weird in this context. Up to you, but I'd consider just replacing the colon with a period:
```suggestion
* Before this update, the collector was discarding longer audit log messages with the following error message: *Internal log [Found line that exceeds max_line_bytes; discarding.]*. With this update, the discarding of longer audit messages is avoided by increasing the audit configuration thresholds. The maximum line size, `max_line_bytes`, is `3145728` bytes. The maximum number of bytes read during a read cycle, `max_read_bytes`, is `262144` bytes. (link:https://issues.redhat.com/browse/LOG-6379[LOG-6379])
```

Author: Thank you for the suggestion, but to the best of my knowledge, this is a legit punctuation use case.

* Before this update, an input receiver's service was repeatedly created and deleted, leading to issues with mounting TLS secrets. With this update, the service is created once and only deleted if it is not defined in the specification. (link:https://issues.redhat.com/browse/LOG-6383[LOG-6383])

* Before this update, pipeline validation could enter an infinite loop if one name was a substring of another. With this update, stricter name equality checks prevent the infinite loop. (link:https://issues.redhat.com/browse/LOG-6405[LOG-6405])

* Before this update, the collector-alerting rules included summary and message fields. With this update, the collector alerting rules now include summary and description fields. (link:https://issues.redhat.com/browse/LOG-6407[LOG-6407])

> 

Author: Missing?

https://issues.redhat.com/browse/LOG-6420

* Before this update, setting up the custom audit inputs in the `ClusterLogForwarder` custom resource with configured `LokiStack` output caused errors due to the nil pointer dereference. With this update, the Operator performs the nil checks, preventing such errors. (link:https://issues.redhat.com/browse/LOG-6449[LOG-6449])

> * Before this update, the `ValidLokistackOTLPOutputs` condition appeared in the status of the `ClusterLogForwarder` custom resource even for non-`LokiStack` outputs. With this update, the `ValidLokistackOTLPOutputs` condition is removed, and the validation messages for the existing output conditions are corrected. (link:https://issues.redhat.com/browse/LOG-6469[LOG-6469])

Reviewer 2: Not worth holding up the merge. But, seeing that the PR is on hold:

```suggestion
* Before this update, the `ValidLokistackOTLPOutputs` condition appeared in the status of the `ClusterLogForwarder` custom resource even when the output type is not LokiStack. With this update, the `ValidLokistackOTLPOutputs` condition is removed, and the validation messages for the existing output conditions are corrected. (link:https://issues.redhat.com/browse/LOG-6469[LOG-6469])
```

Author: TY. I applied the suggested change in the first sentence but left the second sentence as is because `are corrected` (=`have been corrected`) is the intended meaning.

* Before this update, the collector did not correctly mount the `/var/log/oauth-server/` path, which prevented the collection of the audit logs. With this update, the volume mount is added, and the audit logs are collected as expected. (link:https://issues.redhat.com/browse/LOG-6484[LOG-6484])

> * Before this update, the `must-gather` script of the {clo} might fail to gather the LokiStack data. With this update, the `must-gather` script is fixed, and the LokiStack data is gathered reliably. (link:https://issues.redhat.com/browse/LOG-6498[LOG-6498])

Reviewer: Should this be framed in past tense?
```suggestion
* Before this update, the `must-gather` script of the {clo} might have failed to gather the LokiStack data. With this update, the `must-gather` script is fixed, and the LokiStack data is gathered reliably. (link:https://issues.redhat.com/browse/LOG-6498[LOG-6498])
```

Author: Thank you for spotting this. Updating.

[id="logging-release-notes-6-1-1-CVEs_{context}"]
== CVEs

> * link:https://access.redhat.com/security/cve/CVE-2024-2511[CVE-2024-2511]

Author: Yes, thank you for the heads up. I already updated the PR against the errata.
